x-pack/plugins/observability/public/pages/slos/components/slos_grid/grid_size.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFormRow, EuiSelect } from '@elastic/eui';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+
+export function GridSize({ setGridSize }: { setGridSize: (gridSize?: string) => void }) {
+  const [value, setValue] = useLocalStorage('slo-view-grid-size', '3');
+
+  useEffect(() => {
+    setGridSize(value);
+  }, [setGridSize, value]);
```

Comment:
We could just call the `onChangeGridSize` from the EuiSelect onChange, and let the parent component handles storing the value in the localStorage.

Reply:
that will not trigger the change of component actually, since local storage changes doens't cause side effect